### PR TITLE
DOC: Add My week 2 blog post

### PR DESCRIPTION
**Pull request by WassCodeur**
Add my week 2 blog post

### CODE REVIEW

File: docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Hello again,
+~~~~~~~~~~~~~
+
+Welcome back to my Google Summer of Code (GSoC) 2024 journey! This week has been dedicated to refining and improving the work done so far, with a particular focus on the keyword_only decorator.
+
+
+Renaming and Updating the Decorator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This week, I renamed the keyword_only decorator to warn_on_args_to_kwargs for greater clarity. The updated decorator now includes version parameters from_version and until_version. This enhancement ensures that the decorator will raise a RuntimeError if the current version of FURY is greater than until_version.
```
Comment (robinroy03) [inline docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst:18]: Using text formatting will be good when mentioning code snippets.

ex:
warn_on_args_to_kwargs -> `warn_on_args_to_kwargs`

File: docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+Peer Code Review
+~~~~~~~~~~~~~~~~~
+
+I also spent time reviewing `Kaustav Deka's <https://github.com/deka27>`_ code. This exercise remains rewarding, as it helps me understand different coding styles and approaches. Constructive feedback and suggestions from my classmates were invaluable, not only in helping my teammates but also in improving my own coding and reviewing skills.
```
Comment (robinroy03) [inline docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst:24]: could you link the specific PR you reviewed

File: docs/source/posts/2024/2024-06-06-week1-wachiou-bouraima.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
+WEEK 1: Progress and challenges at Google Summer of Code (GSoC) 2024
```
Comment (robinroy03) [inline docs/source/posts/2024/2024-06-06-week1-wachiou-bouraima.rst:1]: Your week 1 blog was merged earlier. Are you planning an edit here?

**Review by robinroy03 — CHANGES_REQUESTED** [review]
Hi, nice work :) Please see the below suggestions.

File: docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst (hunk 1 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst:18]: Hi @robinroy03

Thank you for yout feedback. Okay

File: docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst (hunk 2 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-06-15-week2-wachiou-bouraima.rst:24]: Sure

File: docs/source/posts/2024/2024-06-06-week1-wachiou-bouraima.rst (hunk 3 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-06-06-week1-wachiou-bouraima.rst:1]: No, I'm not planning an edit, it seems that the problem is due to an inclusion of the history of previous modifications.

**Pull request closed without merging**